File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l=0;
        int r=1;
        int maxP = 0;
        while(r<prices.size()){
            if(prices[r] > prices[l]){
                maxP = max(prices[r]-prices[l], maxP);
            }
            if(prices[r] < prices[l]){
                l = r;
            }
            r++;
        }
        return maxP;
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        maxDepth = 0
        def helper(node, depth):
            nonlocal maxDepth
            maxDepth = max(maxDepth, depth)
            if not node:
                return
            helper(node.left, depth+1)
            helper(node.right, depth+1)
        helper(root, 0)
        return maxDepth